src/application/jobs/notifier-actualisation.job.handler.db.ts

```diff
@@ -0,0 +1,159 @@
+import { Inject, Injectable } from '@nestjs/common'
+import { Op } from 'sequelize'
+import { Job } from '../../building-blocks/types/job'
+import { JobHandler } from '../../building-blocks/types/job-handler'
+import { structuresRappelActualisation } from '../../domain/core'
+import {
+  Notification,
+  NotificationRepositoryToken
+} from '../../domain/notification/notification'
+import {
+  Planificateur,
+  PlanificateurRepositoryToken,
+  ProcessJobType
+} from '../../domain/planificateur'
+import { SuiviJob, SuiviJobServiceToken } from '../../domain/suivi-job'
+import { JeuneSqlModel } from '../../infrastructure/sequelize/models/jeune.sql-model'
+import { DateService } from '../../utils/date-service'
+
+interface Stats {
+  nbNotifsEnvoyees: number
+  estLaDerniereExecution: boolean
+  reprogrammmationPourLeLendemain: boolean
+}
+
+const PAGINATION_NOMBRE_DE_JEUNES_MAXIMUM = 17000
+
+@Injectable()
+@ProcessJobType(Planificateur.JobType.NOTIFIER_ACTUALISATION)
+export class NotifierActualisationJobHandler extends JobHandler<Job> {
+  constructor(
+    @Inject(NotificationRepositoryToken)
+    private notificationRepository: Notification.Repository,
+    @Inject(SuiviJobServiceToken)
+    suiviJobService: SuiviJob.Service,
+    private dateService: DateService,
+    @Inject(PlanificateurRepositoryToken)
+    private planificateurRepository: Planificateur.Repository
+  ) {
+    super(Planificateur.JobType.NOTIFIER_ACTUALISATION, suiviJobService)
+  }
+
+  async handle(
+    job: Planificateur.Job<{
+      offset: number
+      nbNotifsEnvoyees: number
+    }>
+  ): Promise<SuiviJob> {
+    let succes = true
+    const stats: Stats = {
+      nbNotifsEnvoyees: job.contenu.nbNotifsEnvoyees,
+      estLaDerniereExecution: false,
+      reprogrammmationPourLeLendemain: false
+    }
+    const maintenant = this.dateService.now()
+    const heure = maintenant.hour
+    const jour = maintenant.weekday
+    const jeudi = 4
+
+    const heureMinuteExecution = {
+      hour: 8,
+      minute: 0,
+      second: 0,
+      millisecond: 0
+    }
+
+    if (jour === jeudi || heure >= 18) {
+      const dateExecution = maintenant
+        .plus({ days: 1 })
+        .setZone('Europe/Paris')
+        .set(heureMinuteExecution)
+        .toJSDate()
+
+      await this.planificateurRepository.creerJob({
+        dateExecution,
+        type: Planificateur.JobType.NOTIFIER_ACTUALISATION,
+        contenu: job.contenu
+      })
+
+      stats.reprogrammmationPourLeLendemain = true
+
+      return {
+        jobType: this.jobType,
+        nbErreurs: 0,
+        succes,
+        dateExecution: maintenant,
+        tempsExecution: DateService.calculerTempsExecution(maintenant),
+        resultat: stats
+      }
+    }
+
+    try {
+      const offset = job.contenu.offset
+
+      const idsJeunesANotifier = await JeuneSqlModel.findAll({
+        where: {
+          structure: {
```

Comment:
`structure: structuresRappelActualisation` est transformé automatiquement en `IN` nan ?

Reply:
yep